tests/func/test_scm.py

```diff
@@ -61,112 +62,95 @@ def test_commit_in_submodule(self):
+    entry, gitignore = scm._get_gitignore(link)
+    assert entry == "/link"
+    assert gitignore == fspath(tmp_dir / ".gitignore")
 
-        foo = os.path.join(repo_dir._root_dir, repo_dir.FOO)
-        bar = os.path.join(repo_dir._root_dir, repo_dir.BAR)
-        gitignore = os.path.join(repo_dir._root_dir, Git.GITIGNORE)
 
-        git.ignore(foo)
+def test_get_gitignore_subdir(tmp_dir, scm):
+    tmp_dir.gen({"dir1": {"file1": "cont", "dir2": {}}})
 
-        with open(gitignore, "r") as fobj:
-            last = fobj.readlines()[-1]
+    data_dir = fspath(tmp_dir / "dir1" / "file1")
+    entry, gitignore = scm._get_gitignore(data_dir)
+    assert entry == "/file1"
+    assert gitignore == fspath(tmp_dir / "dir1" / ".gitignore")
 
-        assert last.endswith("\n")
+    data_dir = fspath(tmp_dir / "dir1" / "dir2")
+    entry, gitignore = scm._get_gitignore(data_dir)
+    assert entry == "/dir2"
+    assert gitignore == fspath(tmp_dir / "dir1" / ".gitignore")
 
-        git.ignore(bar)
 
-        with open(gitignore, "r") as fobj:
-            last = fobj.readlines()[-1]
+def test_gitignore_should_end_with_newline(tmp_dir, scm):
+    tmp_dir.gen({"foo": "foo", "bar": "bar"})
 
-        assert last.endswith("\n")
+    foo = fspath(tmp_dir / "foo")
+    bar = fspath(tmp_dir / "bar")
+    gitignore = tmp_dir / ".gitignore"
 
-    def test_gitignore_should_append_newline_to_gitignore(self, git, repo_dir):
-        git = Git(repo_dir._root_dir)
+    scm.ignore(foo)
+    assert gitignore.read_text().endswith("\n")
 
-        foo_ignore_pattern = "/foo"
-        bar_ignore_pattern = "/bar"
-        bar_path = os.path.join(repo_dir._root_dir, repo_dir.BAR)
-        gitignore = os.path.join(repo_dir._root_dir, Git.GITIGNORE)
+    scm.ignore(bar)
+    assert gitignore.read_text().endswith("\n")
 
-        with open(gitignore, "w") as fobj:
-            fobj.write(foo_ignore_pattern)
 
-        with open(gitignore, "r") as fobj:
-            last = fobj.readlines()[-1]
-        assert not last.endswith("\n")
+def test_gitignore_should_append_newline_to_gitignore(tmp_dir, scm):
+    tmp_dir.gen({"foo": "foo", "bar": "bar"})
 
-        git.ignore(bar_path)
+    foo_ignore_pattern = "/foo"
+    bar_ignore_pattern = "/bar"
+    bar_path = fspath(tmp_dir / "bar")
+    gitignore = tmp_dir / ".gitignore"
 
-        with open(gitignore, "r") as fobj:
-            lines = list(fobj.readlines())
+    gitignore.write_text(foo_ignore_pattern)
+    assert not gitignore.read_text().endswith("\n")
 
-        assert len(lines) == 2
-        for l in lines:
-            assert l.endswith("\n")
+    scm.ignore(bar_path)
+    lines = gitignore.read_text().splitlines()
+    assert len(lines) == 2
+    assert gitignore.read_text().endswith("\n")
 
-        assert lines[0].strip() == foo_ignore_pattern
-        assert lines[1].strip() == bar_ignore_pattern
+    assert lines[0].strip() == foo_ignore_pattern
+    assert lines[1].strip() == bar_ignore_pattern
```

Comment:
Won't that work?
```python
assert gitignore.read_text() == "/foo\n/bar\n"
```
Or at least:
```python
assert lines == ["/foo", "/bar"]
```
`.splitlines()` should already do the strip and no need to check for len this way. BTW, I prefer hard-coding strings like `/foo` and `/bar` here as you see, makes a test more obvious until it short enough)

Reply:
> Won't that work?

Not sure,  might not work because of CRLF.